chain/processor_test.go

```diff
@@ -0,0 +1,435 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package chain_test
+
+import (
+	"context"
+	"encoding/binary"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/ava-labs/avalanchego/database/memdb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/trace"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/x/merkledb"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/hypersdk/auth"
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/crypto"
+	"github.com/ava-labs/hypersdk/crypto/ed25519"
+	"github.com/ava-labs/hypersdk/genesis"
+	"github.com/ava-labs/hypersdk/internal/validitywindow"
+	"github.com/ava-labs/hypersdk/internal/validitywindow/validitywindowtest"
+	"github.com/ava-labs/hypersdk/internal/workers"
+	"github.com/ava-labs/hypersdk/state"
+	"github.com/ava-labs/hypersdk/state/metadata"
+	"github.com/ava-labs/hypersdk/utils"
+)
+
+var (
+	_ chain.AuthVM = (*mockAuthVM)(nil)
+
+	heightKey    = string(chain.HeightKey([]byte{0}))
+	timestampKey = string(chain.TimestampKey([]byte{1}))
+
+	errMockVerifyExpiryReplayProtection = errors.New("mock validity window error")
+)
+
+type createBlock func(parentRoot ids.ID) *chain.StatelessBlock
+
+type mockAuthVM struct{}
+
+func (*mockAuthVM) GetAuthBatchVerifier(_ uint8, _ int, _ int) (chain.AuthBatchVerifier, bool) {
+	return nil, false
+}
+
+func (*mockAuthVM) Logger() logging.Logger {
+	panic("unimplemented")
+}
+
+func TestProcessorExecute(t *testing.T) {
+	createValidBlock := func(root ids.ID) *chain.StatelessBlock {
+		block, err := chain.NewStatelessBlock(
+			ids.Empty,
+			time.Now().UnixMilli(),
```

Comment:
nit: avoid non-deterministic behavior in unit tests where possible - this test should pass if we just use `time.Time{}`

Reply:
This test is still inherently flaky, it could be refactored to accept caller-provided timestamps but it's outside of the scope of the PR